e2e/cypress/integration/messaging/pinned_posts_part2_spec.js

```diff
@@ -0,0 +1,257 @@
+    });
+
+    it('MM-T2175 Pin a post in a DM channel Pin a post while viewing empty pinned post RHS', () => {
+        // # Create DM Channel
+        cy.apiCreateDirectChannel([testUser.id, otherUser.id]).then(() => {
+            cy.apiLogin(testUser);
+
+            // # Visit the DM channel
+            cy.visit(`/${testTeam.name}/channels/${testUser.id}__${otherUser.id}`);
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+            // # Post a message
+            cy.postMessage('Hello');
+
+            return cy.getLastPostId();
+        }).then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Now open RHS
+            cy.get(`#post_${postId}`).trigger('mouseover');
+            cy.get(`#CENTER_commentIcon_${postId}`).click({force: true});
+
+            // * Verify that pinned icon is there both in center and RHS
+            cy.get(`#rhsPost_${postId} .post-pre-header`).should('have.text', 'Pinned');
+            cy.get(`#post_${postId} .post-pre-header`).should('have.text', 'Pinned');
+        });
+    });
+
+    it('MM-T2177 Deleting pinned post while viewing RHS pinned posts removes post from RHS', () => {
+        // # Login
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        // # Click pin icon
+        cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+        // * Should not have any pinned posts
+        cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+
+        // # Post a message
+        cy.postMessage('Hello again');
+
+        cy.getLastPostId().then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Now open RHS
+            cy.get(`#post_${postId}`).trigger('mouseover');
+            cy.get(`#CENTER_commentIcon_${postId}`).click({force: true});
+
+            // * Verify that pinned icon is there both in center and RHS
+            cy.get(`#rhsPost_${postId} .post-pre-header`).should('have.text', 'Pinned');
+            cy.get(`#post_${postId} .post-pre-header`).should('have.text', 'Pinned');
+
+            // # Now delete the post
+            cy.get(`#CENTER_button_${postId}`).click({force: true});
+            cy.get(`#delete_post_${postId}`).scrollIntoView().should('be.visible').click();
+            cy.get('#deletePostModalButton').should('be.visible').click();
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+            // * Should not have any pinned posts
+            cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+
+            // * Post should be deleted
+            cy.get(`#post_${postId}`).should('not.exist');
+            cy.get(`#rhsPost_${postId}`).should('not.exist');
```

Comment:
I don't see where "Note: If another user deleted the message, message displays "(message deleted)". Pinned badge still shows, but all disappear on refresh" is fulfilled.

Reply:
I thought it was a note and not a compulsory step to do. Will look into it. Thanks.

Reply:
Yup, you're right. I'm just not sure if it's covered somewhere else. Thanks for adding here!